packages/react-button/src/components/MenuButton/MenuButtonBase.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { compose, mergeProps } from '@fluentui/react-compose';
+import { MenuButtonProps } from './MenuButton.types';
+import { useMenuButton } from './useMenuButton';
+
+export const MenuButtonBase = compose<'button', MenuButtonProps, MenuButtonProps, {}, {}>(
+  (props, ref, options) => {
+    const { state } = options;
+    const { slots, slotProps } = mergeProps(state, options);
```

Comment:
pass generic types for `mergeProps<...>`, so you can actually get typing for slots/slotProps